sqlx-core/src/pool/inner.rs

```diff
@@ -75,6 +75,14 @@ impl<DB: Database> SharedPool<DB> {
         self.is_closed.load(Ordering::Acquire)
     }
 
+    pub fn options(&self) -> &PoolOptions<DB> {
+        &self.options
+    }
+
+    pub fn connect_options(&self) -> &<DB::Connection as Connection>::Options {
+        &self.connect_options
+    }
+
```

Comment:
Adding these to `SharedPool` doesn't do anything. You need to add getters to `Pool` itself in `src/pool/mod.rs`